newm-server/src/main/kotlin/io/newm/server/features/earnings/EarningsRoutes.kt

```diff
@@ -38,19 +39,19 @@ fun Routing.createEarningsRoutes() {
                 earningsRepository.addAll(earnings)
                 respond(HttpStatusCode.Created)
             }
-        }
-        get("$EARNINGS_PATH/{songId}") {
-            // get earnings by song id
-            val songId = parameters["songId"]!!.toUUID()
-            val earnings = earningsRepository.getAllBySongId(songId)
-            respond(earnings)
-        }
-        post("$EARNINGS_PATH/{songId}") {
-            // create earning records for a song based on receiving a total amount of royalties.
-            val songId = parameters["songId"]!!.toUUID()
-            val royaltyRequest: AddSongRoyaltyRequest = receive()
-            earningsRepository.addRoyaltySplits(songId, royaltyRequest)
-            respond(HttpStatusCode.Created)
+            get("{songId}") {
+                // get earnings by song id
+                val songId = parameters["songId"]!!.toUUID()
+                val earnings = earningsRepository.getAllBySongId(songId)
```

Comment:
Here you can use the `ApplicationCall.songId` extension that we use on `SongRoutes`